packages/editor/src/bindings/post-meta.js

```diff
@@ -0,0 +1,47 @@
+/**
+ * WordPress dependencies
+ */
+import { useEntityProp } from '@wordpress/core-data';
+import { select } from '@wordpress/data';
+/**
+ * Internal dependencies
+ */
+import { store as editorStore } from '../store';
+
+const { getCurrentPostType } = select( editorStore );
+
+// Prettify the name until the label is available in the REST API endpoint.
+const keyToLabel = ( key ) => {
+	return key
+		.split( '_' )
+		.map( ( word ) => word.charAt( 0 ).toUpperCase() + word.slice( 1 ) )
+		.join( ' ' );
+};
+
+export default {
+	name: 'post_meta',
+	label: 'Post Meta',
+	component: null,
+	useSource( props, sourceAttributes ) {
+		const { context } = props;
+		const { value: metaKey } = sourceAttributes;
+		const postType = context.postType
+			? context.postType
+			: getCurrentPostType();
+		const [ meta, setMeta ] = useEntityProp(
+			'postType',
+			context.postType,
+			'meta',
+			context.postId
+		);
+
+		if ( postType === 'wp_template' ) {
+			return { placeholder: keyToLabel( metaKey ) };
+		}
+		const metaValue = meta[ metaKey ];
```

Comment:
What should happen when the `metaValue` is empty?

Reply:
Maybe we should always pass a `placeholder` and change the logic [here](https://github.com/WordPress/gutenberg/pull/58085/files/a4dc34a9eaa717c953b0393c82398ec28e9c62c7#diff-99d79bdba99fe4517413b0fd9503eef924e439c03d44504c1ad0606b727cebb6R69-R76) to check if the value is empty. And in that case, use the placeholder instead of the value?

Reply:
That might work better if we always have a fallback available and here we would only have to read the value.

Reply:
Done in [this commit](https://github.com/WordPress/gutenberg/pull/58085/commits/35155385e8fd4099c9f5e48b5042e5f3880ce7e1).